src/library/playlistfeature.cpp

```diff
@@ -19,6 +19,28 @@
 #include "util/dnd.h"
 #include "util/duration.h"
 
+namespace {
+
+QString createPlaylistLable(
+        const QSqlTableModel& playlistTableModel,
+        int row,
+        int nameColumn,
+        int countColumn,
+        int durationColumn) {
+    QString name = playlistTableModel.data(
+        playlistTableModel.index(row, nameColumn)).toString();
+    int count = playlistTableModel.data(
+        playlistTableModel.index(row, countColumn)).toInt();
+    int duration = playlistTableModel.data(
+        playlistTableModel.index(row, durationColumn)).toInt();
+    return QString("%1 (%2) %3").arg(name, QString::number(count),
```

Comment:
I would only extract this line into the function and pass name, count, and duration as arguments. Accessing the columns should be done at the call site and help here.

Reply:
The crate feature already does a better job in mapping the columns, but that was before I knew Diesel. I don't have any motivation of improving any C++ code. Sorry, no more contributions from my side on the database backend until we switch to Rust ;)